controllers/serviceaccount_controller.go

```diff
@@ -202,7 +202,7 @@ func (r ServiceAccountReconciler) Reconcile(_ goctx.Context, req reconcile.Reque
 	// then just return a no-op and wait for the next sync. This will occur when
 	// the Cluster's status is updated with a reference to the secret that has
 	// the Kubeconfig data used to access the target cluster.
-	guestClient, err := r.remoteClientGetter(clusterContext, ProviderServiceAccountControllerName, clusterContext.Client, client.ObjectKeyFromObject(cluster))
+	guestClient, err := r.remoteClusterCacheTracker.GetClient(clusterContext, client.ObjectKeyFromObject(cluster))
 	if err != nil {
 		clusterContext.Logger.Info("The control plane is not ready yet", "err", err)
```

Comment:
```suggestion
		// Requeue if the reconcile failed because the ClusterCacheTracker was locked for
		// the current cluster because of concurrent access.
		if errors.Is(err, remote.ErrClusterLocked) {
			r.Logger.V(5).Info("Requeuing because another worker has the lock on the ClusterCacheTracker")
			return ctrl.Result{Requeue: true}, nil
		}
		clusterContext.Logger.Info("The control plane is not ready yet", "err", err)
```
It's a bit annoying but we have to handle this case (like in core CAPI).

Basically when a workload cluster is down multiple reconciler will try to get the client for the workload cluster. But only one of those reconcilers will actually try to create the client. The others will get the `remote.ErrClusterLocked` error

This code helps to reduce noisy errors in the controller logs

(same in other places where we use it)

Reply:
Done.